src/newgrf_engine.cpp

```diff
@@ -1328,44 +1330,35 @@ static bool EnginePreSort(const EngineID &a, const EngineID &b)
  */
 void CommitVehicleListOrderChanges()
 {
+	/* Build a list of EngineIDs. EngineIDs are sequential from 0 up to the number of pool items with no gaps. */
+	std::vector<EngineID> ordering(Engine::GetNumItems());
+	std::iota(std::begin(ordering), std::end(ordering), 0);
```

Comment:
Too bad we're not a [C++23](https://en.cppreference.com/w/cpp/algorithm/ranges/iota) yet, eh?